### ISSUE TITLE
Using jupysql magic %sql in databricks

### ISSUE THREAD

**Issue by bhishanpdl** [ploomber/jupysql#102]
I have been using the sql magic `%sql` using jupysql module a lot in my local computer. 
Nowadays, I am starting to use the databricks and porting all my jupyter notebooks to databricks.

The databricks already have their implementation of `%sql` and thus, this magic from jupySQL may not work.

Is there any way if can use both `%sql` from databricks and from jupysql?

Such as can use some alias of `%sql` such as `%sqljupy`?

**Comment by edublancas** [comment]
Ah, I never thought about this.

I added `%jupysql`/`%%jupysql` as an alias, but I'm unsure if this will work, because running `%load_ext sql` might overwrite databricks `%sql`/`%%sql`.

please check if this works on databricks by installing it from the dev branch:

```
pip uninstall jupysql
pip install git+https://github.com/ploomber/jupysql@alias
```

```
%load_ext sql
```

And then see if `%sql` runs the databricks magic, and `%jupysql` runs jupysql's magic.

If this doesn't work, we'll have to allow loading the magics with something like `%load_ext jupysql`

**Comment by idomic** [comment]
@bhishanpdl Is there something that the databricks `%sql` client provides that jupysql isn't?

**Comment by bhishanpdl** [comment]
@idomic I do not want to break the existing functionality and use the third-party functionality as an alias so we can get the best of both world. 

Also, %sql from databricks gives pyspark dataframe, however jupysql %sql gives pandas dataframe.
Sometimes we need to work with pyspark dataframe and %sql from databricks is useful.

**Comment by bhishanpdl** [comment]
@edublancas  The suggestion worked.  (confirmed: 2023 Feb 01)


I have a follow up question: Will the branch will be available in the future, or the feature will be incorporated into the new version?

This is great addition, we can do SQL queries using both databricks %sql and jupysql's %jupysql.

I did following things:

```bash
#pip uninstall jupysql
pip install git+https://github.com/ploomber/jupysql@alias

%jupysql sqlite://

df = pd.DataFrame({'A': [10,20,30], 'B': list('abc')})

%jupysql --persist df

%jupysql SELECT * FROM df limit 2;
```

**Comment by edublancas** [comment]
> I have a follow up question: Will the branch will be available in the future, or the feature will be incorporated into the new version?

yes, it will!


thanks for confirming, I'll run a few extra tests on my end and will release this, will be on pip in the next few days!

**Comment by idomic** [comment]
@bhishanpdl if you're up for it, we'll probably need a short tutorial on running queries via both databricks `%sql` and jupysql's `%jupysql`.

**Comment by bhishanpdl** [comment]
@idomic  I figured out how to work with both databricks  `%sql` and jupysql's `%jupysql`.  I will think about sending a PR with a databricks notebook how to use both SQL magics.

**Comment by edublancas** [comment]
we just merged and released this in version  `0.5.4`!
